ports/stm32f4/common-hal/busio/SPI.c

```diff
@@ -67,26 +351,37 @@ void common_hal_busio_spi_unlock(busio_spi_obj_t *self) {
 
 bool common_hal_busio_spi_write(busio_spi_obj_t *self,
         const uint8_t *data, size_t len) {
-    return 0;
+    HAL_StatusTypeDef result = HAL_SPI_Transmit (&self->handle, (uint8_t *)data, (uint16_t)len, 2);
+    if(!(result==HAL_OK)) mp_raise_RuntimeError(translate("SPI write error"));
```

Comment:
```suggestion
    if (result != HAL_OK) {
        mp_raise_RuntimeError(translate("SPI write error"));
    }
```

Here and below please.